src/app/calculator/utilities/electricity-reduction/electricity-reduction-form/electricity-reduction-form.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, OnInit, Output, EventEmitter, Input } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { Settings } from '../../../../shared/models/settings';
+import { ElectricityReductionService, ElectricityReductionResults } from '../electricity-reduction.service';
+import * as d3 from 'd3';
+
+
+@Component({
+  selector: 'app-electricity-reduction-form',
+  templateUrl: './electricity-reduction-form.component.html',
+  styleUrls: ['./electricity-reduction-form.component.css']
+})
+export class ElectricityReductionFormComponent implements OnInit {
+  @Input()
+  form: FormGroup;
+  @Input()
+  settings: Settings;
+  @Input()
+  index: number;
+  @Input()
+  isBaseline: boolean;
+  @Output('emitCalculate')
+  emitCalculate = new EventEmitter<{ form: FormGroup, index: number, isBaseline: boolean }>();
+  @Output('emitChangeField')
+  emitChangeField = new EventEmitter<string>();
+
+  measurementOptions: Array<{ value: number, name: string }>;
+  idString: string;
+
+  individualResults: ElectricityReductionResults;
+  format: any;
+
+  constructor(private electricityReductionService: ElectricityReductionService) { }
+
+  ngOnInit() {
+    this.getFormat();
+    if (this.isBaseline) {
+      this.idString = this.index.toString();
+    }
+    else {
+      this.idString = 'modification_' + this.index;
+    }
+    this.initMeasurementOptions();
+    this.calculate();
+  }
+
+  initMeasurementOptions() {
```

Comment:
If this is basically a static array, just initialize it with the populated values